File: single-language-analysis/dataset-files/python/test_smell_134.py
# Original URL: https://github.com/ACTCollaboration/pyactlike/blob/1cac8c5d047bc2cad991890f2ebf1d8e3fb483b3/test_like.py#L61-L97

@pytest.mark.skip(reason="cobaya optional")
def test_smell_134():
    """Test the Cobaya interface to the ACT likelihood."""
    from cobaya.yaml import yaml_load
    from cobaya.model import get_model

    info_yaml = r"""
        likelihood:
            pyactlike.ACTPol_lite_DR4:
                components: 
                    - tt
                    - te
                    - ee
                lmax: 6000

        theory:
            camb:
                extra_args:
                    lens_potential_accuracy: 1

        params:
            ns:
                prior:
                  min: 0.8
                  max: 1.2
            H0:
                prior:
                  min: 40
                  max: 100       
            yp2:
                prior:
                    min: 0.5
                    max: 1.5       
        """
    info = yaml_load(info_yaml)
    model = get_model(info)
    assert np.isfinite(model.loglike({"ns": 1.0, "H0": 70, "yp2": 1.0})[0])

File: single-language-analysis/dataset-files/java/TestSmell134.java
import org.junit.Test;
import org.junit.Ignore;
import static org.junit.Assert.*;
import cobaya.yaml.YamlLoad;
import cobaya.model.GetModel;

public class TestSmell134 {

    @Ignore("cobaya optional")
    @Test
    public void testCobaya() {
        String infoYaml = "likelihood:\n" +
            "    pyactlike.ACTPol_lite_DR4:\n" +
            "        components: \n" +
            "            - tt\n" +
            "            - te\n" +
            "            - ee\n" +
            "        lmax: 6000\n" +
            "\n" +
            "theory:\n" +
            "    camb:\n" +
            "        extra_args:\n" +
            "            lens_potential_accuracy: 1\n" +
            "\n" +
            "params:\n" +
            "    ns:\n" +
            "        prior:\n" +
            "          min: 0.8\n" +
            "          max: 1.2\n" +
            "    H0:\n" +
            "        prior:\n" +
            "          min: 40\n" +
            "          max: 100\n" +
            "    yp2:\n" +
            "        prior:\n" +
            "            min: 0.5\n" +
            "            max: 1.5";
        Object info = YamlLoad.yamlLoad(infoYaml);
        Object model = GetModel.getModel(info);
        assertTrue(Double.isFinite((double) model.loglike(Map.of("ns", 1.0, "H0", 70, "yp2", 1.0))[0]));
    }
}